### PR TITLE
FF-3380 feat: add EppoClient.Initialized() method to wait for initialization

### PR DESCRIPTION
**Pull request by rasendubi**

### CODE REVIEW

File: eppoclient/configuration.go
Comment (rasendubi) [inline eppoclient/configuration.go:1]: This file was extracted from `configurationstore.go` without any changes

File: eppoclient/client_test.go [hunk 1]
```diff
@@ -290,3 +291,39 @@ func Test_client_correctActionIsReturnedIfBanditLoggerPanics(t *testing.T) {
...
+
+	timedOut := false
+	select {
+	case <-client.Initialized():
+		timedOut = false
+	case <-time.After(1 * time.Millisecond):
+		timedOut = true
+	}
+
+	assert.False(t, timedOut)
```
Comment (leoromanovsky) [inline eppoclient/client_test.go:328]: Big picture question to remind myself how go channels work - assuming the customer is running an HTTP server and has evaluations they want to perform in a variety of handlers with their goal being NOT to perform evaluations until the SDK is initialized, would they add a reference to this channel to a `Context` they pass around and read the channel, or keep a reference to a global eppo SDK and call `client.Initialized`? Do you have a small code example to share?

File: eppoclient/client_test.go [hunk 2]
```diff
@@ -290,3 +291,39 @@ func Test_client_correctActionIsReturnedIfBanditLoggerPanics(t *testing.T) {
...
+func Test_Initialized_timeout(t *testing.T) {
+	var mockLogger = new(mockLogger)
+	client := newEppoClient(newConfigurationStore(), nil, nil, mockLogger, applicationLogger)
+
+	timedOut := false
+	select {
+	case <-client.Initialized():
+		timedOut = false
+	case <-time.After(1 * time.Millisecond):
+		timedOut = true
```
Comment (aarsilv) [inline eppoclient/client_test.go:304]: According to ChatGPT, this select block will wait for one of these two channels to complete, similar to JavaScript's Promise.race() 👌

**Review by aarsilv — APPROVED** [review]
Thanks for doing this!

**Pull request merged**